index.js

```diff
@@ -67,11 +69,12 @@ function traceModuleFunction(module, emit) {
 
         const event = new Event(name);
         for (let i = 0; i !== numInputActions; i++) {
-          const item = inputActions[i];
-          const action = item[0];
-          const params = item[1];
+          const [ action, params ] = inputActions[i];
           action(values, event, params);
         }
+        if (isFunction(spec.callbacks.onEnter)) {
+          spec.callbacks.onEnter(event, this.context);
```

Comment:
why not applying `this` as is - instead of passing only the `context` field ? also passing `args` may be useful, depending on the use case you have in mind for this.